modules/ROOT/pages/_partials/commons/common-logging.adoc

```diff
@@ -29,94 +29,178 @@ include::{root-partials}_block-abstract.adoc[]
+ifeval::["{param-platform}"=="{platform-net}"]
+:logging-classes: {url-api-class-logging}
+endif::[]
+
+ifdef::is-android[]
+.Constraints
+endif::[]
+
+[NOTE]
+--
+The retrieval of logs from the device is out of scope of this feature.
+
+ifdef::is-android[]
+The value returned by `LogLevel.getValue()` is not the Android log level.
+Do not use this API call.
+endif::[]
+--
+
 
 // Begin text block
-Couchbase Lite provides a logging API that unifies the logging behavior across all platforms{fn-2x5}, making debugging and troubleshooting easier during development and in production.
+== Introduction
+Couchbase Lite provides a robust Logging API {fn-2x5} -- see: API References for {logging-classes} -- which make debugging and troubleshooting easier during development and in production.
+It delivers flexibility in terms of how logs are generated and retained, whilst also maintaining the level of logging required by Couchbase Support for investigation of issues.
+
+Log output is split into the following streams:
+
+* <<lbl-file-logs>>
++
+Here logs are written to <<log-file-outputs, separate log files>> filtered by log level, with each log level supporting individual retention policies.
+
+* <<lbl-console-logs>>
++
+--
+You can independently configure and control console logs, which provides a convenient method of accessing diagnostic information during debugging scenarios.
+With console logging, you can fine-tune diagnostic output to suit specific debug scenarios, without interfering with any logging required by Couchbase Support for the investigation of issues.
+--
 
-NOTE: The retrieval of logs from the device is out of scope of this feature.
+* <<lbl-custom-logs>>
++
+--
+For greater flexibility you can implement a custom logging class using the {url-api-iface-log-custom} interface.
 
-Available logging features include:
+--
 
-* Console based logging
-* File based logging
-* Replication logging
-* Custom logging
+In all instances, you control what is logged and at what level using the {url-api-class-log} class.
 
+[#lbl-console-logs]
 == Console based logging
-Default: Enabled.
-
 Console based logging is often used to facilitate troubleshooting during development.
 
+Console logs are your go-to resource for diagnostic information.
+You can easily fine-tune their diagnostic content to meet the needs of a particular debugging scenario, perhaps by increasing the verbosity and-or choosing to focus on messages from a specific domain; to better focus on the problem area.
+
+Changes to console logging are independent of file logging, so you can make change without compromising any files logging streams.
+It is enabled by default.
+To change default settings use database's {url-api-class-log} method to set the required values -- see <<eg-cons-log>>
+
+You will primarily use {url-api-class-log-console-getcons} and {url-api-class-log-console} to control console logging.
+
```

Comment:
I don't know if this is worth noting... although the platform code will log Verbose and Debug logs, in release versions (more or less, the only thing that external users *ever* get), Core will not.  Release Core does not produce any verbose or debug logs: it will log only error, warning and info, regardless of the log-level set in the logger.